langchain/src/retrievers/self_query/index.ts

```diff
@@ -75,29 +77,44 @@ export class SelfQueryRetriever
   }
 
   static fromLLM(
-    opts: QueryConstructorChainOptions & {
-      vectorStore: VectorStore;
-      structuredQueryTranslator: BaseTranslator;
-    }
+    options: QueryConstructorChainOptions &
+      PartialExcept<
+        SelfQueryRetrieverArgs,
+        "vectorStore" | "structuredQueryTranslator"
+      >
   ): SelfQueryRetriever {
-    const { structuredQueryTranslator } = opts;
-    const allowedComparators =
-      opts.allowedComparators ?? structuredQueryTranslator.allowedComparators;
-    const allowedOperators =
-      opts.allowedOperators ?? structuredQueryTranslator.allowedOperators;
-
-    const llmChain = loadQueryConstructorChain({
-      llm: opts.llm,
-      documentContents: opts.documentContents,
-      attributeInfo: opts.attributeInfo,
-      examples: opts.examples,
+    const {
+      structuredQueryTranslator,
       allowedComparators,
       allowedOperators,
-    });
+      llm,
+      documentContents,
+      attributeInfo,
+      examples,
+      vectorStore,
+      llmChain,
```

Comment:
Actually, why don't we just use the constructor directly if we already have the `llmChain` we want to use?

Reply:
Yeah that'll work too, I suppose we'd need to add a section in the documentation explaining how to construct directly though, since it requires the user to use `loadQueryConstructorChain` as the llmChain parameter.